Actions/ReadSettings/ReadSettings.ps1

```diff
@@ -139,13 +141,24 @@ try {
                     $_ -like $getEnvironments -and $_ -notlike '* (PROD)' -and $_ -notlike '* (Production)' -and $_ -notlike '* (FAT)' -and $_ -notlike '* (Final Acceptance Test)'
                 }
             } | Where-Object {
-                Write-Host "Environment: $_"
-                if ($ghEnvironments -contains $_) {
-                    # Environment is GitHub Environment, default branches are controlled on GitHub
-                    $branches = @( '*' )
+                $envName = $_
```

Comment:
Consider extracting this check in a separate function to reduce the complexity of the current statement.

The previous `Where-Object`'s complexity could also be reduced by extracting a function like `IsProdName($evvironmentName)`.

Reply:
Yeah, the right approach here is to pull this out of ReadSettings (like we have done with projects and more) - but doing this refactoring as part of this bug-fix is slightly more than I have time for:-(

Reply:
Agreed. Such refactoring can be done at a later stage. However, it's a good practice to avoid increasing the current code complexity (it will help refactoring later on).

Reply:
Agree, but moving the code from the where-object statement to a function becomes a function with many parameters, which doesn't make sense at all (as this function will ever only be called once), and will cause me to have to do an extensive test on this. Furthermore, this function will be deleted when we do the refactoring later.
IMO - Let's fix the bug without introducing new ones.